test/Google.Api.Gax.Tests/ClientHelperTest.cs

```diff
@@ -60,17 +67,76 @@ public void BuildCallOptions_PopulatesCancellationToken()
         }
 
         [Fact]
+        public void BuildCallOptions_GlobalCallSettings()
+        {
+            var cts = new CancellationTokenSource();
+            CallSettings callSettings = new CallSettings
+            {
+                Headers = new Metadata { new Metadata.Entry("1", "one") },
+                Expiration = Expiration.WithDeadline(new DateTime(78L)),
+                CancellationToken = cts.Token,
+                WriteOptions = new WriteOptions(WriteFlags.BufferHint),
+                PropagationToken = null, // Un-creatable, un-mockable
+                Credentials = null, // Un-creatable, un-mockable
+            };
+            var helper = new ClientHelper(new DummySettings { CallSettings = callSettings });
+            CallOptions callOptions = helper.BuildCallOptions(null, null);
+            Assert.Equal(callSettings.Headers, callOptions.Headers);
+            Assert.Equal(callSettings.Expiration.Deadline.Value, callOptions.Deadline);
+            Assert.Equal(callSettings.CancellationToken, callOptions.CancellationToken);
+            Assert.Equal(callSettings.WriteOptions, callOptions.WriteOptions);
+            Assert.Null(callOptions.PropagationToken);
+            Assert.Null(callOptions.Credentials);
+        }
+
+        [Fact]
         public void BuildCallOptions_AppliesOverride()
         {
             var helper = new ClientHelper(new DummySettings());
-            var modified = new CallOptions(
-                new Metadata { { "foo", "bar" } },
-                DateTime.UtcNow + TimeSpan.FromHours(1), // Just an arbitrary deadline
-                writeOptions: new WriteOptions(WriteFlags.BufferHint)
-            );
-            Func<CallOptions, CallOptions> optionsOverride = _ => modified;
-            var actual = helper.BuildCallOptions(CancellationToken.None, optionsOverride);
-            Assert.Equal(modified, actual);
+            var cts = new CancellationTokenSource();
+            CallSettings callSettings = new CallSettings
+            {
+                Headers = new Metadata { new Metadata.Entry("1", "one") },
+                Expiration = Expiration.WithDeadline(new DateTime(78L)),
+                CancellationToken = cts.Token,
+                WriteOptions = new WriteOptions(WriteFlags.BufferHint),
+                PropagationToken = null, // Un-creatable, un-mockable
+                Credentials = null, // Un-creatable, un-mockable
+            };
+            CallOptions callOptions = helper.BuildCallOptions(null, callSettings);
+            Assert.Equal(callSettings.Headers, callOptions.Headers);
```

Comment:
Headers are now cloned.
Although this Assert.Equal() method is collection-aware, and compares the contents.